src/Symfony/Component/HttpFoundation/Cookie.php

```diff
@@ -62,7 +62,11 @@ public static function fromString($cookie, $decode = false)
         $part = array_shift($parts);
 
         $name = $decode ? urldecode($part[0]) : $part[0];
-        $value = isset($part[1]) ? ($decode ? urldecode($part[1]) : $part[1]) : null;
+        $value = isset($cookie[\strlen($name) + 1]) ? substr($cookie, \strlen($name) + 1, strcspn($cookie, ';', \strlen($name) + 1)) : null;
```

Comment:
Isn't this something that should be fixed in `HeaderUtils::split()`?

Reply:
moved the patch to the `HeaderUtils` class